### PR TITLE
Fix AddOperator() in sato test

### PR DESCRIPTION
**Pull request by y-guyon**
Fixes https://github.com/AOMediaCodec/libavif/issues/2322.

### CODE REVIEW

**Review by vrabaud — APPROVED** [review]

File: tests/gtest/avifsampletransformtest.cc [hunk 1]
```diff
@@ -28,7 +28,7 @@ class AvifExpression : public avifSampleTransformExpression {
     token.type = AVIF_SAMPLE_TRANSFORM_INPUT_IMAGE_ITEM_INDEX;
     token.inputImageItemIndex = inputImageItemIndex;
   }
-  void AddOperator(uint8_t op) {
+  void AddOperator(avifSampleTransformTokenType op) {
```
Comment (wantehchang) [inline tests/gtest/avifsampletransformtest.cc:31]: (This is an unrelated issue.) I noticed that the local variable `token` declared at line 60 shadows the data member `token` of the base class `avifSampleTransformExpression`. It would be good to rename the local variable.

Comment (wantehchang) [inline tests/gtest/avifsampletransformtest.cc:31]: Thank you for the pull request. I am worried that this did not fix the compiler warning. (Compiler warnings are not enabled on tests yet. Sorry about the confusion.) I will test this using a draft pull request of mine later (https://github.com/AOMediaCodec/libavif/pull/2318).

The warning we need to fix is:

```
warning C4244: 'argument': conversion from 'const avifSampleTransformTokenType' to 'uint8_t', possible loss of data
```

Since `token.type` is a uint8_t, the assignment `token.type = op;` is now a conversion from 'const avifSampleTransformTokenType' to 'uint8_t', so I am worried that MSVC will warn about the assignment.

Comment (y-guyon) [reply tests/gtest/avifsampletransformtest.cc:31]: The data member is named `tokens` in `avifSampleTransformExpression`:

https://github.com/AOMediaCodec/libavif/blob/40b5589b003ca11a63572a470ac8744404fec2f9/include/avif/internal.h#L204

So there is no shadowing.

Comment (y-guyon) [reply tests/gtest/avifsampletransformtest.cc:31]: Sorry, it was an oversight on my end. Fixed in https://github.com/AOMediaCodec/libavif/pull/2331.

Comment (wantehchang) [reply tests/gtest/avifsampletransformtest.cc:31]: You are right. I made a mistake when using Code Search, but I don't know what I did wrong :-). Sorry!
